src/nni_manager/core/nnimanager.ts

```diff
@@ -76,6 +77,11 @@ class NNIManager implements Manager {
             status: 'INITIALIZED',
             errors: []
         };
+        this.trialJobMetricListener = (metric: TrialJobMetric) => {
+            this.onTrialJobMetrics(metric).catch((err: Error) => {
+                this.criticalError(new NNIError('Job metrics error', `Job metrics error: ${err.message}`, err));
```

Comment:
In order to handle string type err, please reference this PR to change it like this: ```this.criticalError(NNIError.FromError(err, 'Job metrics error: '));```